### PR TITLE
feat: sturdypost work branch

### PR DESCRIPTION
**Pull request by magik6k**
## Additional Info
This PR is open to make it easier to track progress on SturdyPoSt, and allow for continuous review.

## Checklist

Before you mark the PR ready for review, please make sure that:

- [ ] Commits have a clear commit message.
- [ ] PR title is in the form of of `<PR type>: <area>: <change being made>`
  - example: ` fix: mempool: Introduce a cache for valid signatures`
  - `PR type`: fix, feat, build, chore, ci, docs, perf, refactor, revert, style, test
  - `area`, e.g. api, chain, state, market, mempool, multisig, networking, paych, proving, sealing, wallet, deps
- [ ] If the PR affects users (e.g., new feature, bug fix, system requirements change), update the CHANGELOG.md and add details to the UNRELEASED section.
- [ ] New features have usage guidelines and / or documentation updates in
  - [ ] [Lotus Documentation](https://lotus.filecoin.io)
  - [ ] [Discussion Tutorials](https://github.com/filecoin-project/lotus/discussions/categories/tutorials)
- [ ] Tests exist for new functionality or change in behavior
- [ ] CI is green

### CODE REVIEW

File: miner/miner.go [hunk 1]
```diff
@@ -325,7 +325,7 @@ minerLoop:
 					"block-time", btime, "time", build.Clock.Now(), "difference", build.Clock.Since(btime))
 			}
 
-			if os.Getenv("LOTUS_MINER_NO_SLASHFILTER") != "_yes_i_know_i_can_and_probably_will_lose_all_my_fil_and_power_" {
+			if os.Getenv("LOTUS_MINER_NO_SLASHFILTER") != "_yes_i_know_i_can_and_probably_will_lose_all_my_fil_and_power_" && !build.IsNearUpgrade(base.TipSet.Height(), build.UpgradeWatermelonFixHeight) {
```
Comment (magik6k) [inline miner/miner.go:328]: Seems to be pulled is from calib fix branch, make sure it's either cleaned up, or that that code lands in master first

File: cmd/lotus-provider/stop.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+import (
+	_ "net/http/pprof"
+
+	"github.com/urfave/cli/v2"
+
+	lcli "github.com/filecoin-project/lotus/cli"
+)
+
+var stopCmd = &cli.Command{
```
Comment (magik6k) [inline cmd/lotus-provider/stop.go:11]: Does this command actually call lotus-provider? I believe it doesn't even serve RPC right now.

File: go.mod
Comment (magik6k) [inline go.mod:1]: We pull in 367057 LOC here apparently

**Review by snadrus — COMMENTED** [review]
Adding a few more changes and then all set.

**Review by snadrus — APPROVED** [review]

**Pull request merged**